cloud-functions/estimate_probability_of_change/main.py

```diff
@@ -30,22 +32,38 @@ def calculate_likelyhood(prices, percent, ws):
     percent: percent change user expect to happen
     ws: Window size in term of days
     """
+    if ws < 0:
+       raise ValueError("Time windows can not be negative")
     if len(prices) < ws:
-        percentChange = calculate_percent_chage(prices[0],
-                                                prices[len(prices.size() - 1)])
-        return percentChange >= percent
+        ValueError("There is not enough data to support window size %s", ws)
 
     up = 0
     down = 0
     for i in range(len(prices) - ws + 1):
-        newPercentChange = calculate_percent_chage(prices[i], prices[i+ws - 1])
-        if newPercentChange > percent:
+        newPercentChange = calculate_percent_chage(prices[i], prices[i + ws - 1])
+        if newPercentChange >= percent:
             up = up + 1
         else:
             down = down + 1
+    # flip the operation if request is for negative number
+    ans=(up * 100.0)/(up + down) if percent > 0 else (down * 100.0)/(up + down)
+    return ans
 
-    return (up * 100.0)/(up + down)
+def calculate_batch_likelyhood(prices, percent, ws):
```

Comment:
I am not using this function for now but kept it in case if we want to improve the batch processing. But can delete if you want that way :)